swiss-army-knife/overlaytest.sh

```diff
@@ -1,19 +1,68 @@
 #!/bin/bash
-echo "=> Start network overlay test"
-  kubectl get pods -l name=overlaytest -o jsonpath='{range .items[*]}{@.metadata.name}{" "}{@.spec.nodeName}{"\n"}{end}' | sort -k2 |
-  while read spod shost
-    do kubectl get pods -l name=overlaytest -o jsonpath='{range .items[*]}{@.status.podIP}{" "}{@.spec.nodeName}{"\n"}{end}' | sort -k2 |
-    while read tip thost
-      do
-        if [[ ! $shost == $thost ]]
-        then
-          kubectl --request-timeout='10s' exec $spod -c overlaytest -- /bin/sh -c "ping -c2 $tip > /dev/null 2>&1"
-          RC=$?
-          if [ $RC -ne 0 ]
-            then echo FAIL: $spod on $shost cannot reach pod IP $tip on $thost
-            else echo $shost can reach $thost
-          fi
-        fi
-    done
+
+DNS_CHECK=false
+
+# Parse arguments
+while [[ $# -gt 0 ]]; do
+  case $1 in
+    --dns-check)
+      DNS_CHECK=true
+      shift
+      ;;
+    *)
+      echo "Unknown option: $1"
+      exit 1
+      ;;
+  esac
+done
+
+echo "=> Start network overlay, DNS, and API test"
+
+kubectl get pods -l name=overlaytest -o jsonpath='{range .items[*]}{@.metadata.name}{" "}{@.spec.nodeName}{" "}{@.status.podIP}{"\n"}{end}' | sort -k2 |
```

Comment:
Should the kubectl commands  add `-n default`? For example if using `admin-tools.yaml` (which includes the SA to run these commands from a pod) it’ll create and deploy into it’s own namespace and kubectl won’t select any overlaytest pods